
<html>
<head>
    <TITLE>About SIFT 4G</TITLE>
    <!--<link rel="stylesheet" type="text/css" href="/D:/SIFT4G_WEB/ForSIFT4G_web/SIFT_WEB_4mBII/css/sift.css" />-->
	<link rel="stylesheet" type="text/css" href="/css/sift.css" />
    <link rel="shortcut icon" href="/images/favicon_1.ico">
</head>

<body bgcolor=white>

  <script src="/javascripts/heading_sift4g.js" type="text/javascript"> </script>

  
<TITLE>About SIFT 4G</TITLE>
<H1>

About S<font color=red>I</font>F<font color=blue>T</font> 4G</center>
</H1>



<UL>
<LI><A HREF="#SIFT4G">What is SIFT 4G?</A>
<LI><A HREF="#Methods comparison of SIFT4G and SIFT">Methods Comparison Between SIFT 4G and SIFT</A>
<LI><A HREF="#Performance comparison of SIFT4G and SIFT">Performance Comparison Between SIFT 4G and SIFT</A>

<BR><BR>
<A NAME="SIFT4G"><H2>What is SIFT 4G?</H2></A><A HREF="http://sift.bii.a-star.edu.sg/" >SIFT</A> predicts whether an amino acid substitution affects protein function. It is widely used in bioinformatics, genetics, disease, and mutation studies. 
SIFT 4G is a faster version of SIFT that enables us to scale up and provide SIFT predictions for more organisms (Click <a href ="http://sift-dna.org/sift4g/public/">here</a> to download SIFT databases). It annotates and provides damaging/tolerated predictions for single nucleotide variants. For indels, only annotation is provided. In SIFT 4G, parts of the original SIFT have been implemented on a GPU, so that processing a protein now takes 3 seconds instead of 4 minutes.
That's almost 2 orders of magnitude faster!! 


<BR><BR>
<A NAME="Methods comparison of SIFT4G and SIFT"><H2>Methods Comparison Between SIFT 4G and SIFT</H2></A>
<b><u>Brief Summary</u><BR>
<BR>

SIFT 4G (and SIFT) takes a query sequence and uses multiple alignment information to predict tolerated and deleterious substitutions for every position of the query sequence. <P> 
SIFT 4G (and SIFT) is a multistep procedure that:</b>
									<ol style="margin-left:60px;">
											<li>   searches for similar sequences, </li>
											<li>   chooses closely related sequences that may share similar function to the query sequence , </li>
											<li>   obtains the alignment of these chosen sequences, and </li>
											<li>   calculates normalized probabilities for all possible substitutions from the alignment.  Positions with normalized probabilities less than 0.05 are predicted to be deleterious, those greater than or equal to 0.05 are predicted to be tolerated.<P></li>
									</ol>
<BR>
<div>
<table>
<tr>
<th>SIFT (Click to view <a href="/www/SIFT_help.html#SIFT_PROCEDURE">SIFT help</a>)</th>
<th></th>
<th>SIFT 4G</th>
</tr>
<td class="border" style="width:40%"><b>1. Search a database for homologous sequences </b> 
<BR>
PSI-BLAST (4 minutes)
2 iterations -e 0.0001 -h 0.002
</td>
<td  style="width:10%"></td>
<td class="border" style="width:40%" ><b>1. Search a database for homologous sequences </b><BR>
Use heuristic Smith-Waterman (< 4 seconds)
<ol><li>Get top 5000 hits that have the longest matching subsequence to the query protein.<BR> 
Query protein is broken down into seeds of 5 amino acids, and the seeds are searched against the database. 
All seed hits are extended using the longest increasing subsequence algorithm. Take top 5000 hits.</li>
<li>Top 5000 hits from 1) are aligned by Smith-Waterman algorithm on a GPU with some alignments delegated to CPUs with the help of the <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3777108/">SW# algorithm</a>. The code can be compiled for CPU only, which is intended for users that do not have a CUDA-enabled GPU.</li>
</ol>
</td>
</tr>
<tr> <td class="center" style="width:40%">  <img style="width="30" height="30" src="/images/down-arrow.jpg"/><td style="width:10%">
<td class="center" style="width:0%">  <img style="width="30" height="30" src="/images/down-arrow.jpg"/></tr>
<tr>
<td class="border" style="width:40%"><b>2. Select a subset of homologous sequences that have a certain diversity (30 sec-1 min)</b><BR>
<ol>
<li>Cluster sequences at 90% identity into consensus sequences</li>
<li>Choose a consensus sequence iteratively using PSI-BLAST until median sequence conservation reaches 2.75</li>
</ol>
<td style="width:10%">
<td style="width:40%" class="border"><b>2. Select a subset of homologous sequences that have a certain diversity (< 1 sec)</b><BR>
Choose sequences iteratively using alignments from the first step until median sequence conservation reaches 2.75
</tr>
<tr> <td class="center" style="width:40%">  <img style="width="30" height="30" src="/images/down-arrow.jpg"/>
<td style="width:10%"><td class="center" style="width:40%">  <img style="width="30" height="30" src="/images/down-arrow.jpg"/></tr>
<tr>
<td style="width:40%" class="border"><b>3. Use BLAST alignment of chosen sequences</b>
<td style="width:10%">
<td class="border" style="width:40%"><b>4. Use Smith-Waterman alignment of chosen sequences</b>
</tr>
<tr> <td class="center" style="width:40%">  <img style="width="30" height="30" src="/images/down-arrow.jpg"/>
<td style="width:10%"><td class="center" style="width:40%">  <img style="width="30" height="30" src="/images/down-arrow.jpg"/></tr>
<tr>
<td style="width:40%" class="border">
<b>5.Score substitutions</b>
<!--<b>5. <A HREF="./SIFT_help.html#CALC_PROB">Score substitutions</A></b>-->
<BR>
Calculate probability of amino acid being tolerated based on observed amino acids in alignment and Dirichlet mixtures
<td style="width:10%" >
<td style="width:40%" class="border">
<b>5.Score substitutions</b> 
<!--<b>5. <A HREF="./SIFT_help.html#CALC_PROB">Score substitutions</A></b> -->
(unchanged from previous version)
</tr>
</table>
</div>



<!--next step-->
<br/>
<div>
<BR>
<A NAME="Performance comparison of SIFT4G and SIFT"><H2>Performance Comparison Between SIFT 4G and SIFT</H2></A>
<!--<H3>Performance Comparison Between SIFT 4G and SIFT
[<a href="#" id="toggle_figcomp" value="Show">Show Figures</A>]-->

<!--[<a href="#" id="toggle_figcomp" value="Show">Show Figures</A>]-->
</H3>
<!--<div id="performanceFigure" style="display:none">-->
<table border="1">
<tbody>
<tr>
<td width="50%" class="noBorderBottom">
SIFT and SIFT 4G have similar accuracies and MCC on 4 different datasets. 
</td>
<td width="50%" class="noBorderBottom">
Generally speaking, SIFT 4G has slightly higher specificity, but at the expense of lower sensitivity, compared to SIFT.
</td>
<tr>
<td class="noBorderTop">
<!--<a href="/images/accuracy.png"><img width="49%" src="/images/accuracy.png" /></a>
-->
<a href="/images/SIFT4G_v2.0_Accuracy_MCC.png"><img width="100%" src="/images/SIFT4G_v2.0_Accuracy_MCC.png" /></a>
</td>
<td class="noBorderTop">
<a href="/images/SIFT4G_v2.0_sensitivity_specificity.png"><img  width="100%" src="/images/SIFT4G_v2.0_sensitivity_specificity.png" /></a>
</td>
</tr>
<tr>
<tr>
<td class="noBorderTop" style="vertical-align:text-top;">
<div class="color-box" style="background-color:#ff0000;"></div>HumDiv (Human)<BR>
<div class="color-box" style="background-color:#008000"></div>HumVar (Human)<BR>
<div class="color-box" style="background-color:#ffcc99;"></div>LacI (<i>E.coli</i>)<BR>
<div class="color-box" style="background-color:#3399ff;"></div>Lysozyme (Bacteriophage)<BR>

</td>
<td class="noBorderTop" style="vertical-align:text-top;" >

Common parameters used between SIFT and SIFT 4G: <br>
<ul>
<li>Database: UniRef90 August 4, 2011</li>
<li>Sequence median Info: 2.75</li>
<li>Remove sequences at % identity: 100%</li>
</ul>

<small>Figures by Pauline Ng licensed under <A HREF="https://creativecommons.org/licenses/by/2.0/">CC BY 2.0</A></small></td>
</tr>
</table>
<P>
</div> <!-- end performanceFigure div -->


&nbsp;&nbsp;&nbsp;&nbsp;(And for those who really want to see the raw numbers...) <!-- [<a href="" id="performanceTableClick" value="Show">Show Table</a>]
<div id="performanceTable" style="display:none">-->
<div id="performanceTable"> 
<table border="1">
<thead>
<tr>
<th>Dataset</th>
<th>Algorithm</th>
<th>Accuracy</th>
<th>Precision</th>
<th> Sensitivity</th>
<th>Specificity</th>
<th>Coverage</th>
</tr>
</thead>
<tbody>
<tr>
<td rowspan="3">HumDiv</td>
<td>SIFT</td>
 <td>0.847 (7,694/9,082)
<td>0.758 (2,450/3,233)
<td> 0.802 (2,450/3,055)
<td> 0.870 (5,244/6,027)
<td> 1
</tr>
<tr>
<td>SIFT 4G (v1.0.0)
<td>0.849 (7,707/9,082)
<td> 0.7635 (2,434/3,188)
<td> 0.797 (2,434/3,055)
<td> 0.875 (5,273/6,027)
<td>1
</tr>
<tr>
<td>SIFT 4G (v2.0.0)
<td>0.846 (7,686/9,082)
<td> 0.758 (2,436/3,213)
<td> 0.797 (2,436/3,055)
<td> 0.871 (5,250/6,027)
<td>1
</tr>

<tr><P>
<tr>
<td rowspan="3">HumVar</td>
<td>SIFT
<td> 0.777 (16,455/21,183)
<td>0.821 (10,019/12,210)
<td>0.798 (10,019/12,556)
<td>0.746 (6,436/8,627)
<td>0.9975
</tr>
<tr>
<td>SIFT 4G (v1.0.0)
<td>0.773 (16,420/21,236)
<td>0.815 (10,074/12,366)
<td>0.800 (10,074/12,598)
<td>0.735 (6,346/8,638)
<td>1
</tr>
<tr>
<td>SIFT 4G (v2.0.0)
<td>0.774 (16,445/21,236)
<td>0.814 (10,122/12,437)
<td>0.803 (10,122/12,598)
<td>0.732 (6,323/8,638)
<td>1
</tr>

<tr>
<td rowspan="3">LacI</td>
<td>SIFT
<td>0.711 (2,874/4,041)
<td>0.623 (1,539/2,471)
<td>0.868 (1,539/1,774)
<td>0.589 (1,335/2,267)
<td>1
</tr>
<tr><td>SIFT 4G (v1.0.0)
<td>0.721 (2,913/4,041)
<td>0.639 (1,483/2,320)
<td>0.836 (1,483/1,774)
<td>0.631 (1,430/2,267)
<td>1
</tr>
<tr><td>SIFT 4G (v2.0.0)
<td>0.720 (2,909/4,041)
<td>0.638 (1,486/2,330)
<td>0.838 (1,486/1,774)
<td>0.628 (1,423/2,267)
<td>1
</tr>

<tr><td rowspan="3">T4 lysozyme</td>
<td>SIFT
<td>0.680 (1,371/2,015)
<td>0.497 (521/1,048)
<td>0.817 (521/638)
<td>0.617 (850/1,377)
<td>1
</tr>
<tr><td>SIFT 4G (v1.0.0)
<td>0.708 (1,427/2,015)
<td>0.527 (496/942)
<td>0.777 (496/638)
<td>0.676 (931/1,377)
<td>1
</tr>
<tr><td>SIFT 4G (v2.0.0)
<td>0.708 (1,427/2,015)
<td>0.527 (496/942)
<td>0.777 (496/638)
<td>0.676 (931/1,377)
<td>1
</tr>

</tbody>
</table>
*Version SIFT 5.2.1 is used to assess SIFT performance.<BR>
**HIV-1 protease not shown because there are too many sequences closely related to HIV-1 protease in the database UniRef90, and so there is low specificity for both versions of SIFT.
</div> <!-- end of performance table -->
